code/database.py

```diff
@@ -0,0 +1,87 @@
+from sqlalchemy import create_engine, Column, Integer, Boolean, String, DateTime
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+import datetime
+import hashlib
+from FileSystemObject import File, Directory
+
+
+
+
+engine = create_engine('sqlite:///SyncOrSwimDB.db')
+Session = sessionmaker(bind=engine)
+Base = declarative_base()
+
+class FileObject(Base):
+	__tablename__ = 'files'
+	path = Column(String, primary_key=True)
+	lastModified = Column(DateTime, default=datetime.datetime.utcnow)
+	deleted = Column(Boolean)
+	toEncrypt = Column(Boolean)
+	lastSync = Column(DateTime, default=datetime.datetime.utcnow)
+	encryptedPath = Column(String)
+	
+	def __init__(self, path, lastModified, deleted, toEncrypt, lastSync, encryptedPath):
+		self.path = path
+		self.lastModified = lastModified
+		self.deleted = deleted
+		self.toEncrypt = toEncrypt
+		self.lastSync = lastSync
+		self.encryptedPath = encryptedPath
+
+class DirectoryObject(Base):
+	__tablename__ = 'directories'
+	path = Column(String, primary_key=True)
+	lastModified = Column(DateTime, default=datetime.datetime.utcnow)
+	deleted = Column(Boolean)
+	toEncrypt = Column(Boolean)
+	lastSync = Column(DateTime, default=datetime.datetime.utcnow)
+	children = Column(Integer)
+	
+	def __init__(self, path, lastModified, deleted, toEncrypt, lastSync, children):
+		self.path = path
+		self.lastModified = lastModified
+		self.deleted = deleted
+		self.toEncrypt = toEncrypt
+		self.lastSync = lastSync
+		self.children = children
+
+
+def retrieve(path):
```

Comment:
Ok, so what's happening here is that to python the database.py file is considered a separate module. So when you run import on it it imports the symbol table of database.py (including its functions and variables) into the top modules symbol table under the object "database". I'm not sure if there are any downsides to this or not, so we may just leave it.